javaparser-core-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/InitializePropertyMetaModelsStatementsGenerator.java

```diff
@@ -38,22 +40,27 @@
 import static com.github.javaparser.utils.Utils.decapitalize;
 
 class InitializePropertyMetaModelsStatementsGenerator {
+
     void generate(Field field, ClassOrInterfaceDeclaration nodeMetaModelClass, String nodeMetaModelFieldName, NodeList<Statement> initializePropertyMetaModelsStatements) {
         final AstTypeAnalysis fieldTypeAnalysis = new AstTypeAnalysis(field.getGenericType());
 
         final Class<?> fieldType = fieldTypeAnalysis.innerType;
         final String typeName = fieldType.getTypeName().replace('$', '.');
         final String propertyMetaModelFieldName = field.getName() + "PropertyMetaModel";
-        nodeMetaModelClass.addField("PropertyMetaModel", propertyMetaModelFieldName, PUBLIC);
+
+        FieldDeclaration newField = nodeMetaModelClass.addField("PropertyMetaModel", propertyMetaModelFieldName, PUBLIC);
+        AbstractGenerator.annotateGenerated(newField, this.getClass().getName());
+
         final String propertyInitializer = f("new PropertyMetaModel(%s, \"%s\", %s.class, %s, %s, %s, %s, %s)",
                 nodeMetaModelFieldName,
                 field.getName(),
                 typeName,
                 optionalOf(decapitalize(nodeMetaModelName(fieldType)), isNode(fieldType)),
-                isOptional(field),
-                isNonEmpty(field),
+                this.isOptional(field),
+                this.isNonEmpty(field),
```

Comment:
The next time I change something here, the explicit `this.`'s will suddenly disappear again :-)